torchtext/datasets/ag_news.py

```diff
@@ -25,16 +25,21 @@
     'test': 7600,
 }
 
+OFFSETS = {
+    'train': [0, 2513230, 4989675, 7448280, 9917610, 12373034, 14841577, 17310009, 19755260, 22180771, 24603053, 27027049],
```

Comment:
I'd suggest to use a dictionary that maps line number to bytes.

Reply:
I was curious on reasoning behind using dict over list of bytes. One thing is it would change the look-up logic but performance-wise it would be same right?

Reply:
Yes, it's likely to have the same performance, lists might even be faster, but it's unlikely to show up. For now I'd optimize for correctness.